patches/lua-tree.patch

```diff
@@ -178,12 +178,7 @@ index 625bb0c63..3c7126165 100644
 -local dereference  = vault.get
  local setmetatable = setmetatable
  local getmetatable = getmetatable
--local re_match     = ngx.re.match
--local re_find      = ngx.re.find
-+local re_match     = require "go.re2".match
- local tostring     = tostring
- local concat       = table.concat
- local insert       = table.insert
+ local re_match     = ngx.re.match
```

Comment:
The line after `+  local re_match     = ngx.re.match` isn't removed or changed; it isn't reflected in the diff since it is 3 lines after the last changed line. By way of example:

```
-- a/init.lua
local tablex       = require "pl.tablex"
local pretty       = require "pl.pretty"
local utils        = require "kong.tools.utils"
local cjson        = require "cjson"
local vault        = require "kong.pdk.vault".new()


local is_reference = vault.is_reference
local dereference  = vault.get
local setmetatable = setmetatable
local getmetatable = getmetatable
local re_match     = ngx.re.match
local re_find      = ngx.re.find
local tostring     = tostring
local concat       = table.concat
local insert       = table.insert
```

```
-- b/init.lua
local tablex       = require "pl.tablex"
local pretty       = require "pl.pretty"
local utils        = require "kong.tools.utils"
local cjson        = { array_mt = {} } --- TODO(hbagdi) XXX analyze the impact
local uuid         = require "go.uuid".generate

local setmetatable = setmetatable
local getmetatable = getmetatable
local re_match     = ngx.re.match
local re_find      = ngx.re.find
local tostring     = tostring
local concat       = table.concat
local insert       = table.insert
```
<img width="656" alt="image" src="https://user-images.githubusercontent.com/5423475/186481406-7cd1105b-a237-4841-89a2-fe579f794192.png">


I think there is some confusion being caused because this is a diff of a file containing a list diffs. Take another look at what's being added and removed from `lua-tree.patch` (ie changes in my commit) vs what each diff contained in  `lua-tree.patch` is indicating should be added/removed (ie what the outcome of running `patch -p1 < patches/lua-tree.patch`).
<img width="452" alt="image" src="https://user-images.githubusercontent.com/5423475/186480949-2cb25660-142d-4222-b66e-81a3cf71895c.png">

Reply:
That is the init.lua file that is checked in, not the diff between the 2.8.0 Kong Gateway OSS files in the luarock.

```diff
diff goks/patches/lua-tree.patch goks-orig/patches/lua-tree.patch
40c40
< - Use gluare instead of ngx.re
---
> - Use go.re2 instead of ngx.re
52c52
< - use gluare::match instead of string.match
---
> - use go.re2::match instead of string.match
64c64
< - Use gluare instead of ngx.re
---
> - Use go.re2 instead of ngx.re
167c167
< @@ -1,12 +1,9 @@
---
> @@ -1,16 +1,12 @@
181c181,186
<  local re_match     = ngx.re.match
---
> -local re_match     = ngx.re.match
> -local re_find      = ngx.re.find
> +local re_match     = require "go.re2".match
>  local tostring     = tostring
>  local concat       = table.concat
>  local insert       = table.insert
304c309
< +local match = ngx.re.match
---
> +local match = require "go.re2".match
413,414c418,425
< @@ -30,30 +30,8 @@ local fmt           = string.format
<  local re_match      = ngx.re.match
---
> @@ -33,34 +26,12 @@ local fmt           = string.format
>  local find          = string.find
>  local gsub          = string.gsub
>  local split         = pl_stringx.split
> -local re_find       = ngx.re.find
> -local re_match      = ngx.re.match
> +local re_find       = require "go.re2".find
> +local re_match      = require "go.re2".match
```

Here you can see that there are more entries is the original patch then the current.

Reply:
What are you confused about?

Reply:
I'm sorry, I don't understand. I've looked at this several different ways and validated the desired contents of `lua-tree.patch` against the corresponding files in `lua-tree/share`.